cpp/src/IceStorm/Scanner.l

```diff
@@ -1,64 +1,76 @@
-%{
-
+%top{
 //
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
+#include "IceUtil/ScannerConfig.h"
+#include <cstdint>
+
+}
+
+%{
+
 #include "Ice/Ice.h"
 #include "Parser.h"
 #include "Grammar.h"
 
-#if defined(_MSC_VER)
-// '<' : signed/unsigned mismatch
-#   pragma warning(disable:4018)
-// 'initializing' : conversion from '__int64' to 'int', possible loss of data
-#   pragma warning(disable:4244)
-
-//
-// '=' : conversion from 'size_t' to 'int', possible loss of data
-// The result of fread() is a size_t and gets inserted into an int
-//
-#   pragma warning(disable:4267)
-#endif
-
-#if defined(__GNUC__)
-#   pragma GCC diagnostic ignored "-Wsign-compare"
+#ifdef _MSC_VER
+#    ifdef yywrap
```

Comment:
Why is this required, should we add a comment?

Reply:
I just copy-pasted it because it's always been there.

I'll do some research and see if I can figure out why we have it.

Reply:
Seems like we don't actually need this anymore.
Setting the `%noyywrap` option does this same thing but cleaner.

All this is about what the scanner does when it hits EOF.
This function returns `0` to signal it should wait, then load more. Or `1` to signal "this is truly EOF".